airflow-core/tests/unit/models/test_deadline.py

```diff
@@ -103,6 +105,50 @@ def test_add_deadline(self, dagrun, session):
         assert result.callback == deadline_orm.callback
         assert result.callback_kwargs == deadline_orm.callback_kwargs
 
+    @pytest.mark.parametrize(
+        "conditions",
+        [
+            pytest.param({}, id="empty_conditions"),
+            pytest.param({Deadline.dagrun_id: INVALID_RUN_ID}, id="no_matches"),
+            pytest.param({Deadline.dagrun_id: RUN_ID}, id="single_condition"),
+            pytest.param({Deadline.dagrun_id: RUN_ID, Deadline.dag_id: DAG_ID}, id="multiple_conditions"),
+            pytest.param(
+                {Deadline.dagrun_id: RUN_ID, Deadline.dag_id: INVALID_DAG_ID}, id="mixed_conditions"
+            ),
+        ],
+    )
+    @mock.patch("sqlalchemy.orm.Session")
+    def test_resolve_deadlines(self, mock_session, conditions):
```

Comment:
```suggestion
    def test_remove_deadlines(self, mock_session, conditions):
```

The code under test seems to be the remove right? Same for below

Reply:
`test_resolve_deadlines_invalid_column` for sure is no longer needed once I change that exception block as discussed above.   I think `test_resolve_deadlines` should still be valid though, no?

Reply:
Ah, nevermind, I misread your comment, you were suggesting a rename of the test, not that the code was removed :sweat_smile:    Yeah, I can rename that.  The other one is still getting dropped though

Reply:
Yeah, rename was the suggestion for sure! I should have been more clear :pray: